python/src/com/jetbrains/python/psi/PyUtil.java

```diff
@@ -549,7 +549,10 @@ public static PsiElement resolveToTheTop(@NotNull final PsiElement elementToReso
         break;
       }
       final PsiElement resolve = reference.resolve();
-      if ((resolve == null) ||  checkedElements.contains(resolve) || resolve.equals(currentElement) || !inSameFile(resolve, currentElement)) {
+      if ((resolve == null) ||
```

Comment:
I'd remove unnecessary parentheses here so the condition will be left as one line